Kitty tutorial (Substrate & Frontend side) Edit (#79)
substrate-developer-hub/substrate-how-to-guides: docs/09-tutorials/01-Kitties/overview.md

### What we're building

In Part 1 of this tutorial, we intentionally keep things simple so that you can decide on how you
would like to improve your Substrate Kitties chain later on. For the purposes of what we are
building, Kitties really can only do the following things:

:smiley_cat: &nbsp; Be created either by some original sources or by being bred using existing Kitties.

:smirk_cat: &nbsp; Be sold at a price set by their owner.

:pouting_cat: &nbsp; Be transferred from one owner to another.

**In Part 1, we'll go over:**

> TK!!

Reviewer: ```suggestion
```

Reviewer: Was this a note to self? :)

Author: yes, it is.

1. [**Basic setup**](/docs/Tutorials/Kitties/Part%201/basic-setup) - we will need to spin up a
Substrate node and create a custom pallet.

2. [**Runtime storage**](/docs/Tutorials/Kitties/Part%201/basic-setup) - we will need a total of
three storage items in our pallet to keep track of the amount of Kitties, their owners, and a `Kitty`
object containing a single Kitty's information.

3. [**Dispatchable functions**](/docs/Tutorials/Kitties/Part%201/dispatchables-and-events) - we will
need a total of 5 dispatchable functions: `create`, `set_price`, `transfer`, `buy_kitty`, and
`breed_kitty`.

4. [**Private functions**](/docs/Tutorials/Kitties/Part%201/create-kitties) - we will write a few

> helper functions to handle the heavy lifting logics, amd

Reviewer: ```suggestion
helper functions to handle the heavy lifting logic, and
```

5. [**Helper functions**](/docs/Tutorials/Kitties/Part%201/interacting-functions) - we will write
two helpers functions for our dispatchable functions: `mint` and `transfer_from`.

**In Part 2, we will:**

1. [**Create a UI for interacting with our pallet's storage items**](/docs/Tutorials/Kitties/Part%202/kitties-frontend)

> - we will connect to Substrate Front-end Template and create the UI for each dispatchable calls.

Reviewer: ```suggestion
- we will connect to the Substrate Front-end Template and create UI components for each dispatchable call.
```

2. [**Enhance our UX with Polkadot JS API**](/docs/Tutorials/Kitties/Part%202/kitties-frontend). 

:::info
In any case, if you are stuck in following the tutorial, you can always refer back to the full

> working source code at the following:

Reviewer: ```suggestion
working source code in the following locations:
```

- On Substrate side, it is under [Substrate Node Template repository `tutorials/kitties` branch](https://github.com/substrate-developer-hub/substrate-node-template/tree/tutorials/kitties).
Most of the code changes are under `/pallets/kitties/src/lib.rs`.

> - On the front end side, it is under [Substrate Front-end Template repository `tutorials/kitties` branch](https://github.com/substrate-developer-hub/substrate-front-end-template/tree/tutorials/kitties).

Reviewer: ```suggestion
- On the front end side, it is under the [Substrate Front-end Template repository `tutorials/kitties` branch](https://github.com/substrate-developer-hub/substrate-front-end-template/tree/tutorials/kitties).
```
